### PR TITLE
hypre: Fix problem when using Intel OneAPI mkl static lib

### PR DESCRIPTION
**Pull request by he-b**

### CODE REVIEW

**Comment by osborn9** [comment]
My understanding is that the blas/lapack provider should provide the correct entries for `spec['lapack'].libs`, `spec['blas'].libs`, so then this is something that should be addressed in their packages. 

Then in the hypre package we could use the following (instead of the previous use of `names` and `directories`). 
```
            '--with-lapack-lib=%s' % ' '.join(spec['lapack'].libs),
            '--with-blas-lib=%s' % ' '.join(spec['blas'].libs)
```

**Comment by he-b** [comment]
@osborn9 thank you for your comments. #27742 fixes this issue at the blas/lapack provider package. This change is not needed anymore.

Thanks

**Comment by he-b** [comment]
Reopening, `spec['lapack'].libs` only provides the list of libraries. When using static libraries we need some additional flags.  `spec['lapack'].libs.ld_flags` provides those. If not here, then where in the package should be added?

**Comment by he-b** [comment]
Reopening, `spec['lapack'].libs` only provides the list of libraries. When using static libraries we need some additional flags.  `spec['lapack'].libs.ld_flags` provides those. If not here, then where in the package should be added?

File: var/spack/repos/builtin/packages/hypre/package.py [hunk 1]
```diff
@@ -121,10 +121,8 @@ def configure_args(self):
 
         configure_args = [
             '--prefix=%s' % prefix,
-            '--with-lapack-libs=%s' % ' '.join(lapack.names),
-            '--with-lapack-lib-dirs=%s' % ' '.join(lapack.directories),
-            '--with-blas-libs=%s' % ' '.join(blas.names),
-            '--with-blas-lib-dirs=%s' % ' '.join(blas.directories)
+            '--with-lapack-lib=%s' % ' '.join(lapack.ld_flags),
+            '--with-blas-lib=%s' % ' '.join(blas.ld_flags),
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hypre/package.py:125]: Are you sure you want to use `join()` here?  

From the pipeline tests it appears that at least `blas.ld_flags` is a string and `' '.join(blas.ld_flags)` is adding a space between each character.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/hypre/package.py:125]:
```
/home/software/spack/[padded-to-512-chars]/linux-ubuntu18.04-x86_64/gcc-7.5.0/openmpi-4.1.2-rgqsfvi6ozylzlwgl3h5zkdxlhygq7wd/bin/mpicc -shared  -o libHYPRE-2.23.0.so   /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/IJ_mv/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/distributed_ls/Euclid/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/distributed_ls/ParaSails/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/distributed_ls/pilut/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/distributed_matrix/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/krylov/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/matrix_matrix/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/multivector/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/parcsr_ls/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/parcsr_mv/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/parcsr_block_mv/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/seq_mv/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/sstruct_ls/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/sstruct_mv/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/struct_ls/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/struct_mv/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/utilities/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/blas/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/lapack/*.o /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/IJ_mv/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/parcsr_ls/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/parcsr_mv/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/seq_mv/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/sstruct_ls/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/sstruct_mv/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/struct_ls/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/struct_mv/*.obj /tmp/root/spack-stage/spack-stage-hypre-2.23.0-ei4ralceispb4pl6bijgca3hufpea7dt/spack-src/src/utilities/*.obj     - L / h o m e / s o f t w a r e / s p a c k / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a / l i n u x - u b u n t u 1 8 . 0 4 - x 8 6 _ 6 4 / g c c - 7 . 5 . 0 / o p e n b l a s - 0 . 3 . 1 8 - 6 f t 6 d 7 e s b t 5 r g p 4 d z g 5 4 7 t m d k k o e 5 7 z 4 / l i b - l o p e n b l a s  - L / h o m e / s o f t w a r e / s p a c k / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a c k _ p a t h _ p l a c e h o l d e r _ _ / _ _ s p a / l i n u x - u b u n t u 1 8 . 0 4 - x 8 6 _ 6 4 / g c c - 7 . 5 . 0 / o p e n b l a s - 0 . 3 . 1 8 - 6 f t 6 d 7 e s b t 5 r g p 4 d z g 5 4 7 t m d k k o e 5 7 z 4 / l i b - l o p e n b l a s -lm           -Wl,-soname,libHYPRE-2.23.0.so -Wl,-z,defs 
```

**Comment by alalazo** [comment]
Seems outdated by other PRs that were merged to `develop` in the meanwhile.

**Pull request closed without merging**